mage_ai/authentication/providers/okta.py

```diff
@@ -0,0 +1,85 @@
+import urllib.parse
+import uuid
+from typing import Dict, Optional
+
+import aiohttp
+from aiohttp import BasicAuth
+
+from mage_ai.authentication.oauth.constants import OAUTH_PROVIDER_OKTA
+from mage_ai.authentication.providers.oauth import OauthProvider
+from mage_ai.authentication.providers.sso import SsoProvider
+from mage_ai.authentication.providers.utils import get_base_url
+from mage_ai.settings.sso import OKTA_CLIENT_ID, OKTA_CLIENT_SECRET, OKTA_DOMAIN_URL
+
+
+class OktaProvider(SsoProvider, OauthProvider):
+    provider = OAUTH_PROVIDER_OKTA
+
+    def __init__(self):
+        self.hostname = OKTA_DOMAIN_URL
```

Comment:
Let's validate the hostname and provide good exception message when it's empty or invalid.